### PR TITLE
docs: add overview page for performance section

### PR DESCRIPTION
**Pull request by bencodezen**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: [Issue #56783 · angular/angular](https://github.com/angular/angular/issues/56783)

## What is the new behavior?

Add performance page

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: adev/src/content/guide/performance/overview.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+That said, please note that these best practices will only take the performance of your application so far. At the end of the day, we encourage you to measure performance in order to best understand what custom optimizations are best for your application.
+
+| Guides Types                              | Description                                                                                                |
+| :---------------------------------------- | :--------------------------------------------------------------------------------------------------------- |
+| [Deferrable views](/defer)                | Defer loading of select dependencies within a template.                                                    |
+| [Image optimization](/image-optimization) | Use the `NgOptimizedImage`directive to adopt best practices for loading images.                            |
+| [Server-side rendering](/ssr)             | Learn how to leverage rendering pages on the server to improve load times.                                 |
+| [Build-time prerendering](/prerendering)  | Also known as static-side generation (SSG), is alternate rendering method to improve load times.           |
```
Comment (JeanMeche) [inline adev/src/content/guide/performance/overview.md:14]: Do we miss a word here ? 
```suggestion
| [Build-time prerendering](/prerendering)  | Also known as static-side generation (SSG), is an alternate rendering method to improve load times.           |
```

File: adev/src/content/guide/performance/overview.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+| [Hydration](/hydration)                   | A process to improve application performance by restoring server-side rendered applications on the client. |
```
Comment (JeanMeche) [inline adev/src/content/guide/performance/overview.md:15]: > by restoring server-side rendered applications on the client. 
Does the phrase feel ok to you @AndrewKushnir  ?

File: adev/src/content/guide/performance/overview.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+Learn about different ways you can optimize the performance of your application.
+</docs-decorative-header>
+
+One of the top priorities of any developer is ensuring that their application is as performant as possible. These guides are here to help you follow best practices for building performant applications.
+
+That said, please note that these best practices will only take the performance of your application so far. At the end of the day, we encourage you to measure performance in order to best understand what custom optimizations are best for your application.
+
+| Guides Types                              | Description                                                                                                |
+| :---------------------------------------- | :--------------------------------------------------------------------------------------------------------- |
+| [Deferrable views](/defer)                | Defer loading of select dependencies within a template.                                                    |
```
Comment (AndrewKushnir) [inline adev/src/content/guide/performance/overview.md:11]: nit: people may know this feature as "@defer", so including it into a description would be helpful for those purposes.
```suggestion
| [Deferrable views](/defer)                | Defer loading of select dependencies within a template by wrapping corresponding parts in a `@defer` block.                                                    |
```

File: adev/src/content/guide/performance/overview.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+| [Image optimization](/image-optimization) | Use the `NgOptimizedImage`directive to adopt best practices for loading images.                            |
```
Comment (AndrewKushnir) [inline adev/src/content/guide/performance/overview.md:12]:
```suggestion
| [Image optimization](/image-optimization) | Use the `NgOptimizedImage` directive to adopt best practices for loading images.                            |
```

File: adev/src/content/guide/performance/overview.md (hunk 2 above)
Comment (AndrewKushnir) [inline adev/src/content/guide/performance/overview.md:15]: I'd propose the phrase it a bit differently:
```suggestion
| [Hydration](/hydration)                   | A process to improve application performance by restoring its state after server-side rendering and reusing existing DOM structure as much as possible. |
```

**Comment by bencodezen** [comment]
Sorry for the delay team. Appreciate the suggestions with rephrasing @AndrewKushnir and @JeanMeche!

**Review by JeanMeche — APPROVED** [review]
Thanks for this !

**Review by AndrewKushnir — APPROVED** [review]
@bencodezen thanks for addressing the feedback 👍

**Comment by pkozlowski-opensource** [comment]
This PR was merged into the repository by commit 8df13e1caee35fd9d5daaf2960674e48a32ed8bf.

The changes were merged into the following branches: main, 18.1.x

**Pull request closed without merging**